cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -303,14 +303,18 @@ rebuildInstallPlan verbosity
 
           (projectConfig, projectConfigTransient) <- phaseReadProjectConfig
           localPackages <- phaseReadLocalPackages projectConfig
-          compilerEtc   <- phaseConfigureCompiler projectConfig
+          compilerEtc@(_, _, progdb)
+                        <- phaseConfigureCompiler projectConfig
           _             <- phaseConfigurePrograms projectConfig compilerEtc
+          pkgConfigDB   <- getPkgConfigDb verbosity progdb
```

Comment:
On balance I think this should not be moved out of `phaseRunSolver` and so instead of being passed in, it should be returned as an additional result. This means it will be cached more aggressively (as it is now).

    (solverPlan,
     pkgconfigDB) <- phaseRunSolver ...